Psr/Log/LoggerInterface.php

```diff
@@ -22,104 +22,104 @@ interface LoggerInterface
     /**
      * System is unusable.
      *
-     * @param string  $message
+     * @param string|\Stringable $message
      * @param mixed[] $context
      *
      * @return void
      */
-    public function emergency($message, array $context = array());
+    public function emergency(string|\Stringable $message, array $context = []);
```

Comment:
I'm not sure that I like having `Stringable` included here. It makes implementation more complex, and IMO doesn't add any value. If someone wants to pass a `Stringable`, they can use `message: (string) $object`.

Reply:
Shouldn't this have a `: void` return type?

Reply:
Ah, I see #77

Reply:
The spec says __toString objects are explicitly allowed.  

> Every method accepts a string as the message, or an object with a __toString() method. Implementors MAY have special handling for the passed objects. If that is not the case, implementors MUST cast it to a string.

Not allowing Stringable would be a spec change.  If an implementation right now doesn't handle stringable objects, then it's already out of spec and should be fixed.

Reply:
This is one case where automatic type casting would actually be useful... ah well.

Reply:
automatic type casting is not the same, as it would not allow loggers to do special handling of some Stringable objects if they always get a string